converter/lookml/lookml.py

```diff
@@ -0,0 +1,87 @@
+from itertools import takewhile
+import os
+import lkml
+from typing import Iterator, List, Optional
+
+from dbt_semantic_interfaces.parsing.dir_to_model import SemanticManifestBuildResult
+from dbt_semantic_interfaces.implementations.semantic_model import PydanticSemanticModel
+from dbt_semantic_interfaces.type_enums.time_granularity import TimeGranularity
+from dbt_semantic_interfaces.implementations.semantic_manifest import (
+    PydanticSemanticManifest,
+)
+from dbt_semantic_interfaces.implementations.semantic_model import NodeRelation
+
+from tqdm import tqdm
+
+from converter.lookml.dimension import to_sl_dimensions
+from converter.lookml.measure import to_sl_measures
+from converter.lookml.model import LkmlModel
+
+DEFAULT_TIME_GRANULARITY = TimeGranularity.DAY
+
+
+def lookml_to_semantic_manifest(lookml_project_dir: str) -> SemanticManifestBuildResult:
+    all_semantic_models = []
+    for path in tqdm(list(lookml_file_paths(lookml_project_dir))):
+        with open(path, "r") as f:
+            lookml_model_dict = lkml.load(f)
+            lookml_model = LkmlModel.parse_obj(lookml_model_dict)
+            semantic_models = parse_model(lookml_model)
+            all_semantic_models.extend(semantic_models)
+    manifest = PydanticSemanticManifest(semantic_models=all_semantic_models, metrics=[])
+    return SemanticManifestBuildResult(semantic_manifest=manifest)
+
+
+def lookml_file_paths(lookml_project_dir: str) -> Iterator[str]:
+    for dirpath, dnames, fnames in os.walk(lookml_project_dir):
+        for f in fnames:
+            if f.endswith(".lkml"):
+                yield os.path.join(dirpath, f)
+
+
+def parse_model(lookml_model: LkmlModel) -> List[PydanticSemanticModel]:
+    semantic_models = []
+
+    if not lookml_model.views:
+        return []
+    for view in lookml_model.views:
+        measures = to_sl_measures(view.measures)
+        dimensions = to_sl_dimensions(view.dimensions)
+        alias = get_alias(view.sql_table_name)
+        node_relation = NodeRelation(
```

Comment:
I don't totally understand what `NodeRelation` is used for, but is it ok that all of it's attributes might be blank?
Also wondering why `"schema_name"` and `"relation_name"` would be hardcoded instead of left as empty strings.

Reply:
I also am unsure what `NodeRelation` is used for other than the `alias`. I included the blank/hard-coded attributes to satisfy the interface. They don't seem to have an effect on the yaml spec output. I would prefer to leave them as-is instead of empty strings. That way it is easier to track down if they do end up in the yaml for some reason.

Reply:
Alias is the part that ends up in the `ref` in the yaml output, so we need that. For instance, [this](https://github.com/dbt-labs/looker/blob/9d37876006bf20cc479b9315f01d1e2015acfec4/marketing/ad_reporting.view.lkml#L3) part gets mapped to [here](https://github.com/dbt-labs/looker/pull/790/files#diff-00ae350c38c9d1dffe3583625a378f5d8e96b809f04783da2e251add2e5f7678R84).